crates/polars-ops/src/series/ops/is_between.rs

```diff
@@ -0,0 +1,34 @@
+use std::ops::BitAnd;
+
+use polars_core::prelude::*;
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, Default)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum ClosedInterval {
+    #[default]
+    Both,
+    Left,
+    Right,
+    None,
+}
+
+pub fn is_between(
+    s: &Series,
+    lower: &Series,
+    upper: &Series,
+    closed: ClosedInterval,
+) -> PolarsResult<BooleanChunked> {
+    let left_cmp_op = match closed {
+        ClosedInterval::None | ClosedInterval::Right => Series::gt,
+        ClosedInterval::Both | ClosedInterval::Left => Series::gt_eq,
+    };
+    let right_cmp_op = match closed {
+        ClosedInterval::None | ClosedInterval::Left => Series::lt,
+        ClosedInterval::Both | ClosedInterval::Right => Series::lt_eq,
+    };
+    let left = left_cmp_op(s, lower)?;
+    let right = right_cmp_op(s, upper)?;
+    Ok(left.bitand(right))
```

Comment:
Instead of doing three passes (one for the lower, one for the upper, one for the binary AND), can we instead do a single pass?

Reply:
I'm not sure I know if this can be done, let alone how to do it. Can you please give more details or even better some guidance here?

Reply:
@petrosbar Never mind, it would require more infrastructure work than would be appropriate for this PR. I'll perhaps get to it in the future.

Reply:
@orlp great! Would be nice if you could clear the requested changes status, unless of course there are other changes that you think need to be addressed, so that this PR can proceed.